src/Microsoft.Health.Blob/Registration/DefaultBlobDataStoreConfiguration.cs

```diff
@@ -0,0 +1,22 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Blob.Configs;
+using Microsoft.Health.Blob.Features.Storage;
+
+namespace Microsoft.Health.Blob.Registration
+{
+    internal class DefaultBlobDataStoreConfiguration : IPostConfigureOptions<BlobDataStoreConfiguration>
+    {
+        public void PostConfigure(string name, BlobDataStoreConfiguration options)
```

Comment:
asp.net already manages the instances for us, can we use that instead?

Reply:
Will this change independently of asp.net maintained one when we use IOPtionSnapshot?

Reply:
This is one of the hooks I was reading up on to configure options when they're resolved. I've removed this class and instead used a helpful utility method instead, but the logic is the same. I configure a delegate to modify the options every time they're resolved. So for `IOptions<T>` it happens once, but for `IOptionsSnapshot` this will happen every request as the service has a `Scoped` lifetime.

These configurations shouldn't mess with how the options extensions lib resolves options. In fact, the underlying factory that creates the options looks for these configurations. I was reading about it here: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-5.0#options-interfaces